src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -235,6 +236,51 @@ struct aie_shim_info : aie_metadata
   }
 };
 
+struct kds_cu_stat
+{
+  using result_type = query::kds_cu_stat::result_type;
+  using data_type = query::kds_cu_stat::data_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type)
+  {
+    auto edev = get_edgedev(device);
+
+    using tokenizer = boost::tokenizer< boost::char_separator<char> >;
+    std::vector<std::string> stats;
+    std::string errmsg;
+
+    // The kds_custat_raw is printing in formatted string of each line
+    // Format: "%d,%s:%s,0x%lx,0x%x,%lu"
+    // Using comma as separator.
+    edev->sysfs_get("kds_custat_raw", errmsg, stats);
+    if (!errmsg.empty())
+      throw std::runtime_error(errmsg);
+
+    result_type cuStats;
+    for (auto& line : stats) {
```

Comment:
Maybe precede line 261 with a comment showing an example of what is being parsed.